lib/ruby_lsp/rubocop/runtime_adapter.rb

```diff
@@ -18,12 +18,7 @@ def reload_config
       end
 
       def run_diagnostic(uri, document)
-        @runtime.offenses(
-          uri_to_path(uri),
-          document.source,
-          document.encoding,
-          prism_result: prism_result(document)
-        )
+        @runtime.offenses(uri_to_path(uri), document.source, prism_result: prism_result(document))
```

Comment:
I don't think ignoring the document encoding for the ruby lsp addon is correct. They explicitly pass us an encoding, why should we ignore that? For the addon, the routes never get called, so we have no idea what the encoding actually is unless we look at it from the document they provide.

Do note that the addon is missing tests for different encodings.

Reply:
As LSP is a JSON-RPC based protocol, I would expect source code to always be transmitted in UTF-8. I'm not very familiar with RubyLSP, but could `document.encoding` ever be something other than UTF-8?